File: week1/ContainsDuplicate/ContainsDuplicate.cs
using System;
using System.Collections.Generic;

public class Solution {
    public static bool ContainsDuplicate(int[] nums) {
        HashSet<int> set = new HashSet<int>();

        foreach (int num in nums)
        {
            if (!set.Add(num))
            {
                return true;
            }
        }

        return false;
    }

    public static void Main()
    {
        // Örnek 1
        int[] nums1 = {1, 2, 3, 1};
        Console.WriteLine($"Example 1: {ContainsDuplicate(nums1)}");

        // Örnek 2
        int[] nums2 = {1, 2, 3, 4};
        Console.WriteLine($"Example 2: {ContainsDuplicate(nums2)}");

        // Örnek 3
        int[] nums3 = {1, 1, 1, 3, 3, 4, 3, 2, 4, 2};
        Console.WriteLine($"Example 3: {ContainsDuplicate(nums3)}");
    }
}

File: week1/8bitbully/containsDuplicate.cpp
// 217. Contains Duplicate
#include <unordered_set>
#include <vector>

using namespace std;

bool containsDuplicate(vector<int> &nums) {
  // hashSet O(n)
  unordered_set<int> hashSet;

  for (const auto &num : nums) {

    if (hashSet.count(num)) {
      return true;
    }

    hashSet.insert(num);
  }

  return false;
}
